crypto/s2n_dhe.c

```diff
@@ -269,7 +277,9 @@ int s2n_dh_compute_shared_secret_as_server(struct s2n_dh_params *server_dh_param
 
     pub_key = BN_bin2bn(( const unsigned char * )Yc.data, Yc.size, NULL);
     notnull_check(pub_key);
-    GUARD(s2n_alloc(shared_key, DH_size(server_dh_params->dh)));
+    int server_dh_params_size = DH_size(server_dh_params->dh);
+    ENSURE_POSIX(server_dh_params_size <= INT32_MAX, S2N_ERR_INTEGER_OVERFLOW);
+    GUARD(s2n_alloc(shared_key, server_dh_params_size));
 
     shared_key_size = DH_compute_key(shared_key->data, pub_key, server_dh_params->dh);
```

Comment:
This could use a `GUARD_OSSL` to capture OSSL error codes.

Reply:
We're already using an error handler here.
```c
    shared_key_size = DH_compute_key(shared_key->data, pub_key, server_dh_params->dh);
    if (shared_key_size <= 0) {
        BN_free(pub_key);
        S2N_ERROR(S2N_ERR_DH_SHARED_SECRET);
    }
```
If I replace it with `GUARD_OSSL`, I'd avoid `BN_free(pub_key)`.

Reply:
I see, we could use `DEFER_CLEANUP` to clean up the pub_key if we exit early in this case.

Reply:
I see, we could use DEFER_CLEANUP to clean up the pub_key if we exit early in this case.

```
DEFINE_POINTER_CLEANUP_FUNC(BIGNUM *, BN_free);
DEFER_CLEANUP(pub_key = BN_bin2bn(( const unsigned char * )Yc.data, Yc.size, NULL), BN_free_function);
GUARD_OSSL(shared_key_size = DH_compute_key(shared_key->data, pub_key, server_dh_params->dh), S2N_ERROR_CODE);
```
Ex:

https://github.com/awslabs/s2n/blob/main/crypto/s2n_ecc_evp.c#L32
https://github.com/awslabs/s2n/blob/main/crypto/s2n_ecc_evp.c#L135

Reply:
We can't implement that now https://github.com/awslabs/s2n/issues/2435.